include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -1250,6 +1238,53 @@ class Radiation : public ISimulationPlugin
 
 } // namespace radiation
 } // namespace plugins
+
+namespace particles
+{
+namespace traits
+{
+    template<
+            typename T_Species,
+            typename T_UnspecifiedSpecies
+    >
+    struct SpeciesEligibleForSolver<
+            T_Species,
+            plugins::radiation::Radiation< T_UnspecifiedSpecies >
+    >
+    {
+    using FrameType = typename T_Species::FrameType;
+
+    // this plugin needs the position, weighting, momentum and momentumPrev1 to run
+    using RequiredIdentifiers = MakeSeq_t<
+            position<>,
+            weighting,
+            momentum,
+            momentumPrev1
+    >;
+
+    using SpeciesHasIdentifiers = typename pmacc::traits::HasIdentifiers<
+            FrameType,
+            RequiredIdentifiers
+    >::type;
+
+    using SpeciesHasMass = typename pmacc::traits::HasFlag<
+            FrameType,
+            massRatio<>
+    >::type;
+
+    using SpeciesHasCharge = typename pmacc::traits::HasFlag<
+            FrameType,
+            chargeRatio<>
+    >::type;
+
+    using type = typename bmpl::and_<
+            SpeciesHasIdentifiers,
+            SpeciesHasMass,
+            SpeciesHasCharge
```

Comment:
The requirements for mass and charge were not there in the original version, however I assume those are actually required.

Reply:
I just included them, because it's physically required and they are used in the kernel. I don't think it hurts to require these identifiers. What do you think @PrometheusPi ?

Reply:
Sure, if they are actually used from the kernel, then it's a good fix.

Reply:
I think that this is a great improvement.